src/devices/MAX31856/MAX31856.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+using System.IO;
+using UnitsNet;
+
+namespace Iot.Device.Max31856
+{
+    /// <summary>
+    /// Documentation https://datasheets.maximintegrated.com/en/ds/MAX31856.pdf
+    /// </summary>
+    public class Max31856 : IDisposable
+    {
+        private readonly byte _thermocoupleType;
+        private SpiDevice _spiDevice;
+
+        #region SpiSettings
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int SpiClockFrequency = 5_000_000;
+
+        /// <summary>
+        /// Set the SPI data flow to MSB First
+        /// </summary>
+        public const DataFlow SpiDataFlow = DataFlow.MsbFirst;
+
+        /// <summary>
+        /// Max31856 SPI Mode
+        /// </summary>
+        public const SpiMode SpiModeSetup = SpiMode.Mode1;
+
+        #endregion
+
+        /// <summary>
+        /// Command to Get Temperature from the device
+        /// </summary>
+        /// <remarks>
+        /// Initializes the device and then reads the data converting the bytes to thermocouple temperature
+        /// </remarks>
+        public Temperature GetTemperature()
+        {
+            Temperature temperature = ReadTemperature(out byte[] spiOutputData);
+            return temperature;
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the Cold-Junction sensor
+        /// </summary>
+        /// <returns>
+        /// Temperature, precision +- 0.7 Celsius range from -20 Celsius to +85 Celsius
+        /// </returns>
+        public Temperature GetColdJunctionTemperature() => Temperature.FromDegreesCelsius(ReadCJTemperature());
+
+        /// <summary>
+        /// Creates a new instance of the max31856.
+        /// </summary>
+        /// <param name="spiDevice">The communications channel to a device on a SPI bus</param>
+        /// <param name="thermocoupleType">Thermocouple Default Type is T can change to B,E,J,K,N,R,S</param>
+        public Max31856(SpiDevice spiDevice, ThermocoupleType thermocoupleType)
```

Comment:
```suggestion
        public Max31856(SpiDevice spiDevice, ThermocoupleType thermocoupleType = ThermocoupleType.T)
```

(the docs say it defaults to T but it doesn't in the code)